### PR TITLE
test-execute: check if nobody exists before running some tests

### PR DESCRIPTION
**Pull request by ronnychevalier**

### CODE REVIEW

File: src/test/test-execute.c [hunk 1]
```diff
@@ -124,11 +144,15 @@ static void test_exec_systemcallerrornumber(Manager *m) {
...
+        if (user_exists("nobody")) {
+                test(m, "exec-user.service", 0, CLD_EXITED);
+        }
 }
 
 static void test_exec_group(Manager *m) {
-        test(m, "exec-group.service", 0, CLD_EXITED);
+        if (group_exists("nobody")) {
+                test(m, "exec-group.service", 0, CLD_EXITED);
+        }
```
Comment (zonque) [inline src/test/test-execute.c:155]: No curly brackets for single-statement block, please.

File: src/test/test-execute.c [hunk 2]
```diff
@@ -145,7 +169,9 @@ static void test_exec_umask(Manager *m) {
 static void test_exec_runtimedirectory(Manager *m) {
         test(m, "exec-runtimedirectory.service", 0, CLD_EXITED);
         test(m, "exec-runtimedirectory-mode.service", 0, CLD_EXITED);
-        test(m, "exec-runtimedirectory-owner.service", 0, CLD_EXITED);
+        if (group_exists("nobody")) {
+                test(m, "exec-runtimedirectory-owner.service", 0, CLD_EXITED);
+        }
```
Comment (zonque) [inline src/test/test-execute.c:174]: Dito.

File: src/test/test-execute.c [hunk 3]
```diff
@@ -29,6 +33,22 @@
...
+        return !!pw;
+}
+
+static bool group_exists(const char *group) {
+        struct group *gr;
+
+        gr = getgrnam(group);
+
+        return !!gr;
+}
```
Comment (zonque) [inline src/test/test-execute.c:50]: Are those wrapper really necessary?

File: src/test/test-execute.c [hunk 4]
```diff
@@ -124,11 +144,15 @@ static void test_exec_systemcallerrornumber(Manager *m) {
 }
 
 static void test_exec_user(Manager *m) {
-        test(m, "exec-user.service", 0, CLD_EXITED);
+        if (user_exists("nobody")) {
```
Comment (zonque) [inline src/test/test-execute.c:147]: `if (getpwnam("nobody"))` is just as readable, ain't it?

Comment (ronnychevalier) [inline src/test/test-execute.c:147]: Yeah... don't know what I was thinking, thanks :)

**Pull request closed without merging**